src/Engine/ProtoCore/BuildStatus.cs

```diff
@@ -26,6 +26,12 @@ public BuildHaltException(string message)
 
     namespace BuildData
     {
+        public enum ErrorID
```

Comment:
Yes it is the error type. However it just follows the current convention in BuildStatus
such as public enum WarningID.

Its easy to modify but theres gonna be hundreds of changes, so i think once you LGTM this, i can then modify the constants so it doesnt litter the review this early?